### PR TITLE
OAuth

### PR DESCRIPTION
**Pull request by volovyks**
Info about OIDC, token validation, and how we are planning to handle internal user id added to README.
Some of the TODOs moved to separate issues.

Not sure how to fix audit CI, chrono crate is up tp date. Can we patch `time` version? Is it a good practice?

### CODE REVIEW

File: mpc-recovery/src/oauth.rs [hunk 1]
```diff
@@ -65,51 +70,154 @@ pub struct TestTokenVerifier {}
...
 
-#[tokio::test]
-async fn test_verify_token_invalid_with_test_verifier() {
-    let token = "invalid";
-    let account_id = TestTokenVerifier::verify_token(token).await;
-    assert_eq!(account_id, None);
-}
+/// This function validates JWT (OIDC ID token) by checking the signature received
+/// from the issuer, issuer, audience, and expiration time.
+pub fn validate_jwt(
```
[the remark this answers is not in the document]
Comment (volovyks) [reply mpc-recovery/src/oauth.rs:97]: Moved to the OAuthTokenVerifier trait.

File: mpc-recovery/src/oauth.rs [hunk 2]
```diff
@@ -65,51 +70,154 @@ pub struct TestTokenVerifier {}
 
 #[async_trait::async_trait]
 impl OAuthTokenVerifier for TestTokenVerifier {
-    async fn verify_token(token: &str) -> Option<&str> {
+    async fn verify_token(token: &str) -> Result<String, String> {
         match token {
             "validToken" => {
                 tracing::info!("TestTokenVerifier: access token is valid");
```
[the remark this answers is not in the document]
Comment (volovyks) [reply mpc-recovery/src/oauth.rs:76]: Is it some kind of a standard?

File: mpc-recovery/src/oauth.rs [hunk 3]
```diff
@@ -65,51 +90,150 @@ pub struct TestTokenVerifier {}
 
 #[async_trait::async_trait]
 impl OAuthTokenVerifier for TestTokenVerifier {
-    async fn verify_token(token: &str) -> Option<&str> {
+    async fn verify_token(token: &str) -> Result<String, String> {
         match token {
             "validToken" => {
-                tracing::info!("TestTokenVerifier: access token is valid");
-                Some("testAccountId")
+                tracing::info!("target: test-token-verifier, access token is valid");
```
Comment (itegulov) [inline mpc-recovery/src/oauth.rs:96]: target value should be outside of quotes 😅 

```suggestion
                tracing::info!(target: "test-token-verifier", "access token is valid");
```

Comment (volovyks) [reply mpc-recovery/src/oauth.rs]: Hah, got it :)

**Pull request merged**